eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -13,49 +7,26 @@ steps:
       $VcpkgCommit = $(Get-Content eng/vcpkg-commit.txt)
       Write-Host "Target Path for vcpkg: $TargetPath"
       Write-Host "Vcpkg SHA: $VcpkgCommit"
+      Write-Host "Cache mode: $env:VCPKG_CACHE_MODE"
+      Write-Host "Cache SAS: $env:VCPKG_CACHE_SAS"
 
       Write-Host "##vso[task.prependpath]$TargetPath"
       Write-Host "##vso[task.setvariable variable=VCPKG_INSTALLATION_ROOT]$TargetPath"
       Write-Host "##vso[task.setvariable variable=VcpkgCommit]$VcpkgCommit"
+      Write-Host "##vso[task.setvariable variable=VCPKG_BINARY_SOURCES;issecret=true]clear;x-azblob,https://cppvcpkgcache.blob.core.windows.net/public-vcpkg-container,$env:VCPKG_CACHE_SAS,$env:VCPKG_CACHE_MODE"
```

Comment:
Is VCPKG_BINARY_SOURCES a well-known variable that VCPKG picks up?

Reply:
Is the SAS token needed in read mode as well? If folks want to use this locally would they need to generate a SAS?

Reply:
Also just so it is clear the SAS will be empty for any public builds as they cannot access that value.

Reply:
Reading a little more at https://vcpkg.io/en/docs/users/binarycaching.html is there reason we chose blob storage over say DevOps Artifact feed or something? I mostly ask because they mark the blob support as expermential.

Reply:
> Is VCPKG_BINARY_SOURCES a well-known variable that VCPKG picks up?

Yes. https://vcpkg.io/en/docs/users/binarycaching.html

> Is the SAS token needed in read mode as well? If folks want to use this locally would they need to generate a SAS?

No, public pipelines, and anyone locally can just set `VCPKG_BINARY_SOURCES=clear;x-azblob,https://cppvcpkgcache.blob.core.windows.net/public-vcpkg-container,,read` to ask VCPKG to pull from there.  

> Also just so it is clear the SAS will be empty for any public builds as they cannot access that value.

Yes, that's the desired behavior :)

Reply:
> Reading a little more at https://vcpkg.io/en/docs/users/binarycaching.html is there reason we chose blob storage over say DevOps Artifact feed or something? I mostly ask because they mark the blob support as experimental.

DevOps and GitHub require Nuget.  We would need to install `Nuget + Mono` for non-windows machines. I didn't like that.
I talked to VCPKG team and confirm that they will only keep adding features to the Azure Storage source ( like auth modes, config, etc ), but it won't be removed.  So I went that route as it won't require `mono` or anything else